src/main/java/net/sf/jabref/gui/specialfields/SpecialFieldIcon.java

```diff
@@ -0,0 +1,84 @@
+package net.sf.jabref.gui.specialfields;
+
+import java.util.Objects;
+
+import javax.swing.Icon;
+import javax.swing.JLabel;
+
+import net.sf.jabref.gui.IconTheme;
+import net.sf.jabref.logic.specialfields.SpecialFieldLocalization;
+import net.sf.jabref.model.entry.specialfields.SpecialField;
+import net.sf.jabref.model.entry.specialfields.SpecialFieldValue;
+
+public class SpecialFieldIcon {
+
+    public static Icon getRepresentingIcon(SpecialField field) {
+        Objects.requireNonNull(field);
+
+        switch (field) {
+            case PRINTED:
+                return IconTheme.JabRefIcon.PRINTED.getSmallIcon();
+            case PRIORITY:
+                return IconTheme.JabRefIcon.PRIORITY.getSmallIcon();
+            case QUALITY:
+                return IconTheme.JabRefIcon.QUALITY.getSmallIcon();
+            case RANK:
+                return IconTheme.JabRefIcon.RANKING.getIcon();
+            case READ_STATUS:
+                return IconTheme.JabRefIcon.READ_STATUS.getSmallIcon();
+            case RELEVANCE:
+                return IconTheme.JabRefIcon.RELEVANCE.getSmallIcon();
+            default:
+                return null;
```

Comment:
Well, the code is actually unreachable, since all enum values are cases. The compiler is just not smart enough to detect it, although it would be feasible. The default case would only be reached if someone added a new SpecialField and forgot to add an Icon mapping. So I would say it does not really matter.